### ISSUE TITLE
[Feature] Reorder New Tab three-dot menu and add navigation

### ISSUE THREAD

**Issue by eliserichards** [mozilla-mobile/fenix#17770]
Meta: #17796
Followup to #17771 

UX point of contact: @violasong (Victoria)

Figma designs (bottom right, labeled "Final design"): https://www.figma.com/file/NHu4cTmzfYgi3QJz5DIDWD/Fenix-Toolbar-Menus?node-id=0%3A1

## Acceptance criteria
This should be the `HomeMenu.kt` file
* **Icons are not being removed. We are waiting on designs to reflect this**
* Reorder the menu items to match designs. From top to bottom:
   * Bookmarks
   * History
   * Downloads
   * Extensions (this is a label change for addons. Followup #17799)
   * Sign in to sync (followup: #17798)
   * Reconnect to Sync
   * Desktop site + toggle (followup: #17800)
   * What's new
   * Help
   * Settings
   * Quit
* Add dividing lines to match designs
* This implementation should add a sticky navigation section to the bottom of the new tab three-dot menu. (Main menu navigation added in #17802)
   * The user should be able to navigate _back_ to the new tab page, then open this three-dot menu to navigate _forwards_ again
   * The back button should be unclickable/grayed out when you are on the new tab page
   * The location of this should match the Top/Bottom orientation of the toolbar setting

---
**Current** (left) vs **new design** (right)
<img width="300" alt="image" src="https://user-images.githubusercontent.com/43795363/106786862-a5aeaf80-6614-11eb-906b-e2d2639a3466.png">. <img width="302" alt="image" src="https://user-images.githubusercontent.com/43795363/106664266-70e61e00-656a-11eb-93cc-36868b3ccd7d.png">

**Comment by kglazko** [comment]
Adding a ticket for UI tests update since this could be a breaking change https://github.com/mozilla-mobile/fenix/issues/17840

**Comment by kglazko** [comment]
@eliserichards @violasong What should we be doing for toggle to desktop in this menu? I only realized while coding it that in the previous PR where this menu is open when a tab is open, we use this `private val selectedSession: TabSessionState? get() = store.state.selectedTab` which makes me wonder what to do when there's no selected tab.

**Comment by eliserichards** [comment]
@kglazko I broke out the implementation of desktop site into another story, if you want to just add the item in this PR: https://github.com/mozilla-mobile/fenix/issues/17800

**Comment by AndRo-Marian** [comment]
Why I need the What's New, Help and Settings to be to bottom?
